src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -147,9 +158,14 @@ private static Solution GetSolutionWithReplacedDocuments(Solution solution, IDoc
         private readonly IDocumentChangeTracker _documentChangeTracker;
 
         /// <summary>
-        /// The solution state that the request should operate on.
+        /// The workspace that the documeny was found in, if any, if <see cref="IRequestHandler{RequestType, ResponseType}.GetTextDocumentIdentifier(RequestType)"/> has a non-null return.
+        /// </summary>
+        public readonly Workspace Workspace;
```

Comment:
I would prefer if this wasn't available to the handlers to use and potentially do crazy things, but I'm not sure the best way of doing that.  Don't know if we could make it private, then have a method that takes in the cache to remove it?

Reply:
Yeah I struggled with this a bit too, because what I really wanted to do was just pass in the solution, but in a way that only the queue can use it, not the handlers. Passing in the cache would work.

Reply:
This cache is `Dictionary<Workspace, (Solution, Solution)>` and I feel bad passing it in as a parameter anywhere, so I went with an out parameter. See what you think.